include/zephyr/net/ieee802154_radio.h

```diff
@@ -31,72 +31,311 @@ extern "C" {
  * @{
  */
 
-/* See section 6.1: "Some of the timing parameters in definition of the MAC are in units of PHY
- * symbols. For PHYs that have multiple symbol periods, the duration to be used for the MAC
- * parameters is defined in that PHY clause."
+/**
+ * MAC functional description (section 6)
+ */
+
+/**
+ * The symbol period (and therefore symbol rate) is defined in section 6.1: "Some
+ * of the timing parameters in definition of the MAC are in units of PHY symbols.
+ * For PHYs that have multiple symbol periods, the duration to be used for the
+ * MAC parameters is defined in that PHY clause."
+ *
+ * This is not necessarily the true physical symbol period, so take care to use
+ * this macro only when either the symbol period used for MAC timing is the same
+ * as the physical symbol period or if you actually mean the MAC timing symbol
+ * period.
+ *
+ * PHY specific symbol periods are defined in PHY specific sections below.
  */
-#define IEEE802154_PHY_SUN_FSK_863MHZ_915MHZ_SYMBOL_PERIOD_US 20U /* see section 19.1, table 19-1 */
-#define IEEE802154_PHY_OQPSK_2450MHZ_SYMBOL_PERIOD_US         16U /* see section 12.3.3 */
+#define IEEE802154_PHY_SYMBOLS_PER_SECOND(symbol_period_ns) (NSEC_PER_SEC / symbol_period_ns)
 
-/* TODO: Get PHY-specific symbol period from radio API. Requires an attribute getter, see
- *       https://github.com/zephyrproject-rtos/zephyr/issues/50336#issuecomment-1251122582.
- *       For now we assume PHYs that current drivers actually implement.
+/**
+ * MAC services (section 8)
```

Comment:
it looks as if you would either want to have this not be a doxygen comment, or instead create `@name`d sections, since I don't think you want "MAC services (section 8)" to only show up for `IEEE802154_MAC_A_BASE_SLOT_DURATION`. This comment applies to basically all the other "(section XXX)" comments.

<img width="773" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/128251/322d071d-9aa0-4507-a8c7-96d69acd15e8">

Reply:
Oh, and if you find similar issues in #62503 then of course it would be great to point me to it as that one really intends to fix all of them as far as possible.

Reply:
> @kartben Is it ok if we postpone all documentation issues until this one is merged, as it will immediately unblock #62503 which fixes this problem and lots and lots of other documentation issues as well?

Absolutely :)